src/app/obfboard.ts

```diff
@@ -162,20 +250,45 @@ export class Sound {
 export class OBFBoard {
 
   format: string;
+
+  @IsNotEmpty({ message: 'Board id must be specified'})
+  @IsString({ message: 'Board id must be a string'})
   id: string;
+
   locale: string;
+
   name: string;
+
+  @IsOptional()
+  @IsString()
   descriptionHtml: string;
+
+  @ValidateNested()
+  @IsDefined()
   grid: Grid;
+
+  @ValidateNested({
+    each: true
+  })
+  @IsDefined()
   buttons: Button[];
+
+  @ValidateNested({
+    each: true
+  })
   images: Image[];
+
+  @ValidateNested({
+    each: true
+  })
   sounds: Sound[];
+
   imageResolver: ImageResolver;
   soundResolver: SoundResolver;
 
   deserialize(input: any): OBFBoard {
     this.format = input.format;
-    this.id = String(input.id);
+    this.id = input.id && String(input.id);
```

Comment:
```suggestion
     this.id = (input.id || input.id === 0) && String(input.id);
```
To be safe with input.id of 0